app/portainer/views/endpoints/edit/endpoint.html

```diff
@@ -219,7 +220,7 @@
               <button
                 type="button"
                 class="btn btn-primary btn-sm !ml-0"
-                ng-disabled="state.actionInProgress || !endpoint.Name || !endpoint.URL || (endpoint.TLS && ((endpoint.TLSVerify && !formValues.TLSCACert) || (endpoint.TLSClientCert && (!formValues.TLSCert || !formValues.TLSKey))))"
+                ng-disabled="state.actionInProgress || !endpoint.Name || !endpoint.URL || (endpoint.TLS && ((endpoint.TLSVerify && !formValues.TLSCACert) || (endpoint.TLSClientCert && (!formValues.TLSCert || !formValues.TLSKey)))) || (formValues.tlsConfig.tls && !formValues.tlsConfig.skipVerify && (!formValues.tlsConfig.caCertFile || !formValues.tlsConfig.certFile || !formValues.tlsConfig.keyFile)) "
```

Comment:
the whole point of using `withFormValidation` is that the react form validation communicates with angularjs. so if on line 75, you'll change the form element to be `<form name="$ctrl.endpointForm">`, you'll be able to change this whole condition to `!$ctrl.endpointForm.$valid`

Reply:
fixed